design/pipeline-artifacts.md

```diff
@@ -15,9 +15,9 @@ Artifacts are distinct from other `resources` types, including `containers`, `re
 ### Schema
 
 ```yaml
-- upload: string | [ string ]  # a minimatch path or list of [minimatch paths](tasks/file-matching-patterns.md) to upload
+- upload: string # a minimatch path or list of [minimatch paths](tasks/file-matching-patterns.md) to upload
```

Comment:
I'm proposing we switch this to `string` only, not `string | [ string ]`.

Reasoning: I'm working on changing the parsing infrastructure of the agent to be schema driven - for extensibility. Ultimately this property maps to a task input, which is a `string` not an array of strings. And we don't have any mechanism in the parser currently to convert an array of strings into a single string. That's not to say we can't add it in the future, I just want to eliminate adding a new concept to solve it right now. And a multi-line string is allowed already.

Reply:
Agreed, let's do it.

Reply:
Yep, this was me not fully understanding YAML. As long as multiple strings can be provided, I'm 👍.

Reply:
@vtbassmatt : Can you add this to the list of things you plan to support in v1 of the YAML. Somehow asking user to provide multi-line string for patterns doesn't feel like a good YAML experience :).

Reply:
I'll add it for discussion. It feels like shades of the same "is there more than one syntax for things?" that we want to avoid except when it's unambiguously better for the vast majority of cases. Multiline input is well-trod ground in our tasks (both from the designer era and now in YAML).